packages/vuetify/src/components/VDataTable/VDataTableHeaders.tsx

```diff
@@ -203,7 +222,78 @@ export const VDataTableHeaders = genericComponent<VDataTableHeadersSlots>()({
       )
     }
 
+    const VDataTableMobileHeaderCell = () => {
+      const headerProps = mergeProps(props.headerProps ?? {} ?? {})
+
+      const displayItems = computed<ItemProps['items']>(() => {
+        return columns.value.filter(column => column?.sortable)
+      })
+
+      return (
+        <VDataTableColumn
+          tag="th"
+          class={[
+            ...headerCellClasses.value,
+            loaderClasses.value,
+          ]}
+          colspan={ headers.value.length + 1 }
+          { ...headerProps }
+        >
+          <div class="v-data-table-header__content">
+            <VSelect
+            chips
+            class={[
+              'my-2',
+              'v-data-table__mobile-sort-select',
+            ]}
+            clearable
+            color="primary"
+            density="default"
+            hide-details
+            items={ displayItems.value }
+            label="Sort By"
+            multiple={ props.multiSort }
+            variant="underlined"
+            onClick:clear={ () => sortBy.value = [] }
+            >
+              {{
+                ...slots,
+                chip: props => (
+                  <VChip
+                    onClick={ props.item.raw?.sortable ? () => toggleSort(props.item.raw) : undefined }
+                    onMousedown={ (e: MouseEvent) => {
+                      e.preventDefault()
+                      e.stopPropagation()
+                    }}
+                  >
+                    { props.item.title }
+                    <VIcon
+                      class={[
+                        'ms-1',
+                        'v-data-table__mobile-sort-icon',
+                      ]}
+                      color={ isSorted(props.item.raw) ? 'on-surface' : 'grey' }
```

Comment:
Color by state should be primarily handled in sass. While on-surface is a valid color, grey is a utility color that could be disabled by the user.